cli/train.go

```diff
@@ -0,0 +1,162 @@
+	labels []string,
+) (string, error) {
+	if err := c.ensureLoggedIn(); err != nil {
+		return "", err
+	}
+	if modelVersion == "" {
+		modelVersion = time.Now().Format("2006-01-02T15-04-05")
+	}
+	modelTypeEnum, ok := mltrainingpb.ModelType_value["MODEL_TYPE_"+strings.ToUpper(modelType)]
+	if !ok || modelTypeEnum == int32(mltrainingpb.ModelType_MODEL_TYPE_UNSPECIFIED) {
+		return "", errors.Errorf("%s must be a valid ModelType, got %s. See `viam data train submit --help` for supported options",
+			trainFlagModelType, modelType)
+	}
+
+	resp, err := c.mlTrainingClient.SubmitTrainingJob(context.Background(),
+		&mltrainingpb.SubmitTrainingJobRequest{
+			Filter: filter, OrganizationId: orgID, ModelName: modelName, ModelVersion: modelVersion,
+			ModelType: mltrainingpb.ModelType(modelTypeEnum), Tags: labels,
+		})
+	if err != nil {
+		return "", errors.Wrapf(err, "received error from server")
+	}
+	return resp.Id, nil
+}
+
+// DataGetTrainingJob is the corresponding action for 'data train get'.
+func DataGetTrainingJob(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	job, err := client.dataGetTrainingJob(c.String(trainFlagJobID))
+	if err != nil {
+		return err
+	}
+	printf(c.App.Writer, "Training job: %s", job)
+	return nil
+}
+
+// dataGetTrainingJob gets a training job with the given ID.
+func (c *viamClient) dataGetTrainingJob(trainingJobID string) (*mltrainingpb.TrainingJobMetadata, error) {
+	if err := c.ensureLoggedIn(); err != nil {
+		return nil, err
+	}
+	resp, err := c.mlTrainingClient.GetTrainingJob(context.Background(), &mltrainingpb.GetTrainingJobRequest{Id: trainingJobID})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Metadata, nil
+}
+
+// DataCancelTrainingJob is the corresponding action for 'data train cancel'.
+func DataCancelTrainingJob(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	id := c.String(trainFlagJobID)
+	if err := client.dataCancelTrainingJob(id); err != nil {
+		return err
+	}
+	printf(c.App.Writer, "Successfully sent cancellation request for training job %s", id)
+	return nil
+}
+
+// dataCancelTrainingJob cancels a training job with the given ID.
+func (c *viamClient) dataCancelTrainingJob(trainingJobID string) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+	if _, err := c.mlTrainingClient.CancelTrainingJob(
+		context.Background(), &mltrainingpb.CancelTrainingJobRequest{Id: trainingJobID}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// DataListTrainingJobs is the corresponding action for 'data train list'.
+func DataListTrainingJobs(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	jobs, err := client.dataListTrainingJobs(c.String(dataFlagOrgID), c.String(trainFlagJobStatus))
+	if err != nil {
+		return err
+	}
+	for _, job := range jobs {
+		printf(c.App.Writer, "Training job: %s\n", job)
```

Comment:
I think the `\n` is redundant in `printf` [[0](https://github.com/viamrobotics/rdk/blob/323caeb590771eb2351ed094fca4b77ebd446985/cli/print.go#L26)]

Reply:
This was for prettification — with a long list, it was visually hard for me to parse between training jobs unless there was an empty newline between them.